ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarGroupDemoController_SwiftUI.swift

```diff
@@ -0,0 +1,179 @@
+                        avatarCount -= 1
+                    }
+                }, label: {
+                    Image(systemName: "minus")
+                })
+                .disabled(avatarCount <= 0)
+                .buttonStyle(AvatarCountButtonStyle())
+
+                Button(action: {
+                    if avatarCount < samplePersonas.capacity - 1 {
+                        avatarCount += 1
+                    }
+                }, label: {
+                    Image(systemName: "plus")
+                })
+                .disabled(avatarCount >= samplePersonas.count - 1)
+                .buttonStyle(AvatarCountButtonStyle())
+            }
+
+            HStack {
+                Text("Max Displayed Avatars")
+                Spacer()
+                TextField("Max Displayed Avatars", value: $maxDisplayedAvatars, formatter: NumberFormatter())
+                    .autocapitalization(.none)
+                    .disableAutocorrection(true)
+                    .multilineTextAlignment(.trailing)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+                    .frame(width: 100)
+            }
+
+            HStack {
+                Text("Overflow Count")
+                Spacer()
+                TextField("Overflow Count", value: $overflowCount, formatter: NumberFormatter())
+                    .autocapitalization(.none)
+                    .disableAutocorrection(true)
+                    .multilineTextAlignment(.trailing)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+                    .frame(width: 100)
+            }
+        }
+    }
+
+    @ViewBuilder
+    var avatarSettings: some View {
+        Group {
+            DemoHeading(title: "Content")
+
+            FluentUIDemoToggle(titleKey: "Set alternate background", isOn: $useAlternateBackground)
+            FluentUIDemoToggle(titleKey: "Transparency", isOn: $isTransparent)
+        }
+
+        Group {
+            DemoHeading(title: "Ring")
+            FluentUIDemoToggle(titleKey: "Ring visible", isOn: $isRingVisible)
+            FluentUIDemoToggle(titleKey: "Ring inner gap", isOn: $hasRingInnerGap)
+            FluentUIDemoToggle(titleKey: "Set image based ring color", isOn: $showImageBasedRingColor)
+        }
+
+        Group {
+            DemoHeading(title: "Size")
+
+            Picker("Size", selection: $size) {
+                Text(".xxlarge").tag(MSFAvatarSize.xxlarge)
+                Text(".xlarge").tag(MSFAvatarSize.xlarge)
+                Text(".large").tag(MSFAvatarSize.large)
+                Text(".medium").tag(MSFAvatarSize.medium)
+                Text(".small").tag(MSFAvatarSize.small)
+                Text(".xsmall").tag(MSFAvatarSize.xsmall)
+            }
+            .frame(maxWidth: .infinity, alignment: .leading)
+        }
+
+    }
+
+    public var body: some View {
+        VStack {
+            AvatarGroup(style: style, size: size, avatarCount: avatarCount) { avatar, index in
```

Comment:
What I had in mind to work with the states in the creation of the AvatarGroup would be something like:

AvatarGroup(style: style,
size: size,
**avatarBuilder: (avatarGroupState: MSFAvatarGroupState) -> void**)

This way the calling code can implement the closure to iterate on any logic it wants and create the initial avatars directly from the avatarGroupState object.

Reply:
So in this version, they'd be responsible for looping and calling `createAvatar()` on the passed-in `MSFAvatarGroupState` directly?

Interesting proposal, but I have two concerns:
1. Something feels a bit "off" about asking people to run procedural/OO method calls in the middle of a declarative statement like this. Mixing of metaphors and all that. I'd ideally prefer something that is purely declarative if at all possible.
2. I'm not sure how I feel about exposing `MSFAvatarGroupState` like this; we've been good so far about not showing our state to our SwiftUI consumers.

I toyed with having the closure yield an array of `MSFAvatarGroupAvatarState` objects, but we don't really have a clean way of yielding these without the `create` method you referenced above.

I feel like we're so close to a great paradigm...